packages/pwa-kit-react-sdk/src/ssr/universal/components/with-react-query/index.js

```diff
@@ -39,7 +39,11 @@ export const withReactQuery = (Wrapped) => {
             const queryClient = (res.locals.__queryClient =
                 res.locals.__queryClient || new QueryClient())
 
-            await ssrPrepass(appJSX)
+            // Without the request object, our useServerContext hook would be able tell whether on prepass
+            const withoutReq = React.cloneElement(appJSX, {
+                req: undefined
+            })
```

Comment:
@bendvc I couldn't set both `req` and `res` to be undefined. Why only setting `req`? Because of a few reasons:
- Our code is relying on the existence of res.locals. For example, see line 25 above.
- I did try a fake `res` with `res: {locals: {}}`. But that has a negative consequence: on the 2nd pass (for line 25), `locals.__queryClient` would have been undefined... it should have been defined already from the prepass.

Reply:
Good catch. I guess this is another good example of why not to use `locals`. This isn't the ideal solution, but it works.